<Type Name="LinqDataSourceValidationException" FullName="System.Web.UI.WebControls.LinqDataSourceValidationException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="bbaddf486803c0dcc0984d6f452e14da42b85436" /><Meta Name="ms.sourcegitcommit" Value="055a4a82a0b08bfbdc21bd1347fb71f7fe2c099e" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="pl-PL" /><Meta Name="ms.lasthandoff" Value="08/15/2019" /><Meta Name="ms.locfileid" Value="69390809" /></Metadata><TypeSignature Language="C#" Value="public class LinqDataSourceValidationException : Exception, System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit LinqDataSourceValidationException extends System.Exception implements class System.Runtime.Serialization.ISerializable, class System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="DocId" Value="T:System.Web.UI.WebControls.LinqDataSourceValidationException" />
  <TypeSignature Language="VB.NET" Value="Public Class LinqDataSourceValidationException&#xA;Inherits Exception&#xA;Implements IDynamicValidatorException" />
  <TypeSignature Language="C++ CLI" Value="public ref class LinqDataSourceValidationException : Exception, System::Web::DynamicData::IDynamicValidatorException" />
  <TypeSignature Language="F#" Value="type LinqDataSourceValidationException = class&#xA;    inherit Exception&#xA;    interface IDynamicValidatorException&#xA;    interface ISerializable" />
  <AssemblyInfo>
    <AssemblyName>System.Web.Extensions</AssemblyName>
    <AssemblyVersion>3.5.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces>
    <Interface>
      <InterfaceName>System.Runtime.Serialization.ISerializable</InterfaceName>
    </Interface>
    <Interface>
      <InterfaceName>System.Web.DynamicData.IDynamicValidatorException</InterfaceName>
    </Interface>
  </Interfaces>
  <Attributes>
    <Attribute>
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="de588-101">Opisuje wyjątek, który wystąpił podczas weryfikacji wartości nowych lub zmodyfikowanych, zanim wartości zostaną wstawione, zaktualizowane lub usunięte przez <see cref="T:System.Web.UI.WebControls.LinqDataSource" /> formant.</span><span class="sxs-lookup"><span data-stu-id="de588-101">Describes an exception that occurred during validation of new or modified values before values are inserted, updated, or deleted by a <see cref="T:System.Web.UI.WebControls.LinqDataSource" /> control.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-102"><xref:System.Web.UI.WebControls.LinqDataSourceValidationException> Klasa zawiera wyjątki konwersji typu i metody ustawiającej właściwość.</span><span class="sxs-lookup"><span data-stu-id="de588-102">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class contains type conversion and property setter exceptions.</span></span> <span data-ttu-id="de588-103">Wszystkie wyjątki, które są zgłaszane podczas walidacji, znajdują się <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> w kolekcji.</span><span class="sxs-lookup"><span data-stu-id="de588-103">All of the exceptions that are thrown during validation are contained in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection.</span></span> <span data-ttu-id="de588-104">Każdy wyjątek weryfikacji można pobrać poprzez iterację <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> kolekcji.</span><span class="sxs-lookup"><span data-stu-id="de588-104">You can retrieve each validation exception by iterating through the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection.</span></span>  
  
 <span data-ttu-id="de588-105">Zazwyczaj obsługiwane są <xref:System.Web.UI.WebControls.LinqDataSource.Updating>wyjątki w zdarzeniach <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, i <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> .</span><span class="sxs-lookup"><span data-stu-id="de588-105">You typically handle exceptions in the <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, and <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> events.</span></span> <span data-ttu-id="de588-106">`Exception` Jeśli wystąpi wyjątek walidacji, właściwość <xref:System.Web.UI.WebControls.LinqDataSourceUpdateEventArgs>obiektu, <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs>lub <xref:System.Web.UI.WebControls.LinqDataSourceInsertEventArgs> zawiera <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> obiekt.</span><span class="sxs-lookup"><span data-stu-id="de588-106">If a validation exception is thrown, the `Exception` property of the <xref:System.Web.UI.WebControls.LinqDataSourceUpdateEventArgs>, <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs>, or <xref:System.Web.UI.WebControls.LinqDataSourceInsertEventArgs> object contains a <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> object.</span></span> <span data-ttu-id="de588-107">Jeśli żaden wyjątek nie zostanie zgłoszony, `Exception` Właściwość zawiera `null`.</span><span class="sxs-lookup"><span data-stu-id="de588-107">If no exception is thrown, the `Exception` property contains `null`.</span></span>  
  
 <span data-ttu-id="de588-108">Aby obsłużyć wyjątki walidacji, Utwórz procedurę <xref:System.Web.UI.WebControls.LinqDataSource.Updating>obsługi <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>dla zdarzenia <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> , lub.</span><span class="sxs-lookup"><span data-stu-id="de588-108">To handle validation exceptions, create a handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating>, <xref:System.Web.UI.WebControls.LinqDataSource.Deleting>, or <xref:System.Web.UI.WebControls.LinqDataSource.Inserting> event.</span></span> <span data-ttu-id="de588-109">W programie obsługi zdarzeń Sprawdź, czy `Exception` właściwość klasy argument zdarzenia zawiera wartość inną niż. `null`</span><span class="sxs-lookup"><span data-stu-id="de588-109">In the event handler, check whether the `Exception` property of the event argument class contains a value other than `null`.</span></span> <span data-ttu-id="de588-110">Jeśli właściwość nie ma wartości null, zgłoszono co najmniej jeden wyjątek walidacji i można pobrać wszystkie wyjątki walidacji <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> we właściwości. `Exception`</span><span class="sxs-lookup"><span data-stu-id="de588-110">If the `Exception` property is not null, one or more validation exceptions were thrown and you can retrieve each validation exception in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property.</span></span>  
  
 <span data-ttu-id="de588-111">Jeśli wartość nie jest zgodna z typem właściwości, może wystąpić wyjątek walidacji.</span><span class="sxs-lookup"><span data-stu-id="de588-111">A validation exception can occur if a value does not match the type of the property.</span></span> <span data-ttu-id="de588-112">Na przykład próba zaktualizowania właściwości Integer przy użyciu znaków nienumerycznych powoduje wyjątek sprawdzania poprawności.</span><span class="sxs-lookup"><span data-stu-id="de588-112">For example, trying to update an integer property by using non-numeric characters causes a validation exception.</span></span> <span data-ttu-id="de588-113">Klasa LINQ to SQL może również zawierać niestandardowe kryteria weryfikacji, które upewnij się, że właściwość zawiera wartość znajdującą się w oczekiwanym zakresie lub wzorcu.</span><span class="sxs-lookup"><span data-stu-id="de588-113">A LINQ to SQL class can also contain customized validation criteria that make sure that the property contains a value that is within an expected range or pattern.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="de588-114">Poniższy przykład pokazuje procedurę obsługi zdarzeń dla <xref:System.Web.UI.WebControls.LinqDataSource.Updating> zdarzenia.</span><span class="sxs-lookup"><span data-stu-id="de588-114">The following example shows an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating> event.</span></span> <span data-ttu-id="de588-115">Wyświetla wszystkie komunikaty o wyjątkach weryfikacji przy użyciu <xref:System.Web.UI.WebControls.Label> kontrolki.</span><span class="sxs-lookup"><span data-stu-id="de588-115">It displays any validation exception messages by using a <xref:System.Web.UI.WebControls.Label> control.</span></span>  
  
```vb  
Protected Sub LinqDataSource_Updating(ByVal sender As Object, _  
        ByVal e As LinqDataSourceUpdateEventArgs)  
    If (e.Exception IsNot Nothing) Then  
        For Each innerException As KeyValuePair(Of String, Exception) _  
               In e.Exception.InnerExceptions  
            Label1.Text &= innerException.Key & ": " & _  
                innerException.Value.Message & "<br />"  
        Next  
        e.ExceptionHandled = True  
    End If  
End Sub  
```  
  
```csharp  
protected void LinqDataSource_Updating(object sender,   
        LinqDataSourceUpdateEventArgs e)  
{  
    if (e.Exception != null)  
    {  
        foreach (KeyValuePair<string, Exception> innerException in   
             e.Exception.InnerExceptions)  
        {  
        Label1.Text += innerException.Key + ": " +   
            innerException.Value.Message + "<br />";  
        }  
        e.ExceptionHandled = true;  
    }  
}  
```  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="de588-116">Inicjuje nowe wystąpienie klasy <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy.</span><span class="sxs-lookup"><span data-stu-id="de588-116">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="de588-117">Inicjuje nowe wystąpienie <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy z komunikatem dostarczonym przez system, który opisuje błąd.</span><span class="sxs-lookup"><span data-stu-id="de588-117">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-118">Komunikat dostarczony przez system, który opisuje błąd, uwzględnia bieżącą kulturę systemu.</span><span class="sxs-lookup"><span data-stu-id="de588-118">The system-supplied message that describes the error takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="de588-119">W poniższej tabeli przedstawiono początkowe wartości właściwości dla wystąpienia <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> klasy.</span><span class="sxs-lookup"><span data-stu-id="de588-119">The following table shows the initial property values for an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class.</span></span>  
  
|<span data-ttu-id="de588-120">Właściwość</span><span class="sxs-lookup"><span data-stu-id="de588-120">Property</span></span>|<span data-ttu-id="de588-121">Wartość</span><span class="sxs-lookup"><span data-stu-id="de588-121">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="de588-122">`null`.</span><span class="sxs-lookup"><span data-stu-id="de588-122">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="de588-123">Zlokalizowany opis dostarczany przez system.</span><span class="sxs-lookup"><span data-stu-id="de588-123">A system-supplied localized description.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="de588-124">Obsługa i zgłaszanie wyjątków</span><span class="sxs-lookup"><span data-stu-id="de588-124">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="de588-125">Komunikat, który opisuje błąd.</span><span class="sxs-lookup"><span data-stu-id="de588-125">The message that describes the error.</span></span></param>
        <summary><span data-ttu-id="de588-126">Inicjuje nowe wystąpienie <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy z określonym komunikatem opisującym błąd.</span><span class="sxs-lookup"><span data-stu-id="de588-126">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-127">W poniższej tabeli przedstawiono początkowe wartości właściwości dla wystąpienia <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> klasy.</span><span class="sxs-lookup"><span data-stu-id="de588-127">The following table shows the initial property values for an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class.</span></span>  
  
|<span data-ttu-id="de588-128">Właściwość</span><span class="sxs-lookup"><span data-stu-id="de588-128">Property</span></span>|<span data-ttu-id="de588-129">Wartość</span><span class="sxs-lookup"><span data-stu-id="de588-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="de588-130">`null`.</span><span class="sxs-lookup"><span data-stu-id="de588-130">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="de588-131">Ciąg komunikatu o błędzie określony w `message`.</span><span class="sxs-lookup"><span data-stu-id="de588-131">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="de588-132">Obsługa i zgłaszanie wyjątków</span><span class="sxs-lookup"><span data-stu-id="de588-132">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected LinqDataSourceValidationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; LinqDataSourceValidationException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="de588-133">Obiekt zawierający dane serializowanego obiektu.</span><span class="sxs-lookup"><span data-stu-id="de588-133">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="de588-134">Informacje kontekstowe o źródle lub miejscu docelowym.</span><span class="sxs-lookup"><span data-stu-id="de588-134">Context information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="de588-135">Inicjuje nowe wystąpienie klasy <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy z serializowanych danych.</span><span class="sxs-lookup"><span data-stu-id="de588-135">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-136">Ten konstruktor jest wywoływany podczas deserializacji w celu odtworzenia obiektu wyjątku, który jest przesyłany przez strumień.</span><span class="sxs-lookup"><span data-stu-id="de588-136">This constructor is called during deserialization to reconstitute the exception object that is transmitted over a stream.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message, System.Collections.Generic.IDictionary&lt;string,Exception&gt; innerExceptions);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Collections.Generic.IDictionary`2&lt;string, class System.Exception&gt; innerExceptions) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String,System.Collections.Generic.IDictionary{System.String,System.Exception})" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerExceptions As IDictionary(Of String, Exception))" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message, System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ innerExceptions);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string * System.Collections.Generic.IDictionary&lt;string, Exception&gt; -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (message, innerExceptions)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerExceptions" Type="System.Collections.Generic.IDictionary&lt;System.String,System.Exception&gt;" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="de588-137">Komunikat, który opisuje błąd.</span><span class="sxs-lookup"><span data-stu-id="de588-137">The message that describes the error.</span></span></param>
        <param name="innerExceptions"><span data-ttu-id="de588-138">Kolekcja wyjątków walidacji.</span><span class="sxs-lookup"><span data-stu-id="de588-138">The collection of validation exceptions.</span></span></param>
        <summary><span data-ttu-id="de588-139">Inicjuje nowe wystąpienie <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy z określonym komunikatem opisującym błąd i kolekcją odwołań do wyjątków wewnętrznych.</span><span class="sxs-lookup"><span data-stu-id="de588-139">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified message that describes the error, and a collection of references to inner exceptions.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public LinqDataSourceValidationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; LinqDataSourceValidationException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.LinqDataSourceValidationException : string * Exception -&gt; System.Web.UI.WebControls.LinqDataSourceValidationException" Usage="new System.Web.UI.WebControls.LinqDataSourceValidationException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="de588-140">Komunikat, który opisuje błąd.</span><span class="sxs-lookup"><span data-stu-id="de588-140">The message that describes the error.</span></span></param>
        <param name="innerException"><span data-ttu-id="de588-141">Wyjątek, który jest przyczyną bieżącego wyjątku.</span><span class="sxs-lookup"><span data-stu-id="de588-141">The exception that is the cause of the current exception.</span></span></param>
        <summary><span data-ttu-id="de588-142">Inicjuje nowe wystąpienie <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> klasy z określonym komunikatem o błędzie i odwołaniem do wewnętrznego wyjątku.</span><span class="sxs-lookup"><span data-stu-id="de588-142">Initializes a new instance of the <see cref="T:System.Web.UI.WebControls.LinqDataSourceValidationException" /> class with a specified error message and a reference to the inner exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-143">Zazwyczaj nie używa się tego konstruktora, ponieważ wyjątki walidacji są przechowywane we <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> właściwości, a nie <xref:System.Exception.InnerException%2A> na właściwości.</span><span class="sxs-lookup"><span data-stu-id="de588-143">You typically will not use this constructor because validation exceptions are stored in the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property, not the <xref:System.Exception.InnerException%2A> property.</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="de588-144">Obsługa i zgłaszanie wyjątków</span><span class="sxs-lookup"><span data-stu-id="de588-144">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="override this.GetObjectData : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; unit" Usage="linqDataSourceValidationException.GetObjectData (info, context)" />
      <MemberType>Method</MemberType>
      <Implements>
        <InterfaceMember>M:System.Runtime.Serialization.ISerializable.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)</InterfaceMember>
      </Implements>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8">
          <AttributeName>System.Security.SecurityCritical</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="de588-145">Obiekt, który przechowuje serializowane dane obiektu dla zgłoszonego wyjątku.</span><span class="sxs-lookup"><span data-stu-id="de588-145">The object that holds the serialized object data for the exception that is being thrown.</span></span></param>
        <param name="context"><span data-ttu-id="de588-146">Informacje kontekstowe o źródle lub miejscu docelowym.</span><span class="sxs-lookup"><span data-stu-id="de588-146">Context information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="de588-147">Pobiera informacje o wyjątku i dodaje go do <see cref="T:System.Runtime.Serialization.SerializationInfo" /> obiektu.</span><span class="sxs-lookup"><span data-stu-id="de588-147">Gets information about the exception and adds it to the <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-148">Metoda dodaje informacje o wyjątku <xref:System.Runtime.Serialization.SerializationInfo> do obiektu, który jest przeznaczony do serializacji. <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData%2A></span><span class="sxs-lookup"><span data-stu-id="de588-148">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.GetObjectData%2A> method adds information about the exception to the <xref:System.Runtime.Serialization.SerializationInfo> object that is targeted for serialization.</span></span> <span data-ttu-id="de588-149">Podczas deserializacji wyjątek jest retworzony z <xref:System.Runtime.Serialization.SerializationInfo> , który jest przesyłany przez strumień.</span><span class="sxs-lookup"><span data-stu-id="de588-149">During deserialization, the exception is reconstituted from the <xref:System.Runtime.Serialization.SerializationInfo> that is transmitted over the stream.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="InnerExceptions">
      <MemberSignature Language="C#" Value="public System.Collections.Generic.IDictionary&lt;string,Exception&gt; InnerExceptions { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Collections.Generic.IDictionary`2&lt;string, class System.Exception&gt; InnerExceptions" />
      <MemberSignature Language="DocId" Value="P:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property InnerExceptions As IDictionary(Of String, Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ InnerExceptions { System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ get(); };" />
      <MemberSignature Language="F#" Value="member this.InnerExceptions : System.Collections.Generic.IDictionary&lt;string, Exception&gt;" Usage="System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions" />
      <MemberType>Property</MemberType>
      <Implements>
        <InterfaceMember>P:System.Web.DynamicData.IDynamicValidatorException.InnerExceptions</InterfaceMember>
      </Implements>
      <AssemblyInfo>
        <AssemblyName>System.Web.Extensions</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Collections.Generic.IDictionary&lt;System.String,System.Exception&gt;</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="de588-150">Pobiera jeden lub więcej wyjątków, które wystąpiły, gdy zostały zweryfikowane nowe lub zmodyfikowane dane.</span><span class="sxs-lookup"><span data-stu-id="de588-150">Gets one or more exceptions that occurred when new or modified data was being validated.</span></span></summary>
        <value><span data-ttu-id="de588-151">Kolekcja zawierająca wyjątki.</span><span class="sxs-lookup"><span data-stu-id="de588-151">A collection that contains the exceptions.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="de588-152"><xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> Kolekcja zawiera wszystkie wyjątki walidacji, które zostały zgłoszone podczas sprawdzania poprawności danych przed operacją Update, INSERT lub DELETE.</span><span class="sxs-lookup"><span data-stu-id="de588-152">The <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> collection contains all the validation exceptions that were thrown during data validation before an update, insert, or delete operation.</span></span> <span data-ttu-id="de588-153">Jeśli wartość nie jest zgodna z typem właściwości, może wystąpić wyjątek walidacji.</span><span class="sxs-lookup"><span data-stu-id="de588-153">A validation exception can occur if a value does not match the type of the property.</span></span> <span data-ttu-id="de588-154">Na przykład jeśli spróbujesz zaktualizować Właściwość Integer przy użyciu znaków nieliczbowych, zostanie zgłoszony wyjątek walidacji.</span><span class="sxs-lookup"><span data-stu-id="de588-154">For example, if you try to update an integer property by using non-numeric characters, a validation exception is thrown.</span></span> <span data-ttu-id="de588-155">Klasa LINQ to SQL może również zawierać niestandardowe kryteria weryfikacji, które upewnij się, że właściwość zawiera wartość znajdującą się w oczekiwanym zakresie lub wzorcu.</span><span class="sxs-lookup"><span data-stu-id="de588-155">A LINQ to SQL class can also contain customized validation criteria that make sure that the property contains a value that is within an expected range or pattern.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="de588-156">Poniższy przykład pokazuje procedurę obsługi zdarzeń dla <xref:System.Web.UI.WebControls.LinqDataSource.Updating> zdarzenia.</span><span class="sxs-lookup"><span data-stu-id="de588-156">The following example shows an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Updating> event.</span></span> <span data-ttu-id="de588-157">Wyświetla wszystkie komunikaty o wyjątkach weryfikacji przy użyciu <xref:System.Web.UI.WebControls.Label> kontrolki.</span><span class="sxs-lookup"><span data-stu-id="de588-157">It displays any validation exception messages by using a <xref:System.Web.UI.WebControls.Label> control.</span></span>  
  
```vb  
Protected Sub LinqDataSource_Updating(ByVal sender As Object, _  
        ByVal e As LinqDataSourceUpdateEventArgs)  
    If (e.Exception IsNot Nothing) Then  
        For Each innerException As KeyValuePair(Of String, Exception) _  
                In e.Exception.InnerExceptions  
          Label1.Text &= innerException.Key & ": " & _   
                  innerException.Value.Message & "<br />"  
        Next  
        e.ExceptionHandled = True  
    End If  
End Sub  
```  
  
```csharp  
protected void LinqDataSource_Updating(object sender,   
        LinqDataSourceUpdateEventArgs e)   
{  
    if (e.Exception != null)  
    {  
        foreach (KeyValuePair<string, Exception> innerException in   
            e.Exception.InnerExceptions)  
        {  
            Label1.Text += innerException.Key + ": " +   
                innerException.Message + "<br />";  
        }  
        e.ExceptionHandled = true;  
    }  
}  
```  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
